packages/web-wallet/src/services/ReadOnlyWalletWrapper.ts

```diff
@@ -0,0 +1,321 @@
+    }
+  }
+
+  /**
+   * Get current address
+   */
+  getCurrentAddress(): AddressInfo | null {
+    if (!this.wallet?.isReady()) {
+      throw new Error('Wallet not initialized');
+    }
+
+    try {
+      const addressInfo: AddressInfoObject = this.wallet.getCurrentAddress();
+      return {
+        address: addressInfo.address,
+        index: addressInfo.index,
+        transactions: 0, // AddressInfoObject doesn't have transactions field
+      };
+    } catch (error) {
+      log.error('Failed to get current address:', error);
+      throw error; // Re-throw instead of returning null
+    }
+  }
+
+  /**
+   * Get address at a specific index
+   */
+  async getAddressAtIndex(index: number): Promise<AddressInfo | null> {
+    if (!this.wallet?.isReady()) {
+      throw new Error('Wallet not initialized');
+    }
+
+    try {
+      const addressStr: string = await this.wallet.getAddressAtIndex(index);
+      return {
+        address: addressStr,
+        index: index,
+        transactions: 0,
+      };
+    } catch (error) {
+      log.error('Failed to get address at index:', error);
+      throw error;
+    }
+  }
+
+  /**
+   * Get all addresses (returns an async generator)
+   */
+  async *getAllAddresses(): AsyncGenerator<AddressInfo> {
+    if (!this.wallet?.isReady()) {
+      throw new Error('Wallet not initialized');
+    }
+
+    try {
+      const addresses = this.wallet.getAllAddresses();
+      for await (const addr of addresses) {
+        yield {
+          address: addr.address,
+          index: addr.index,
+          transactions: addr.transactions,
+        };
+      }
+    } catch (error) {
+      log.error('Failed to get all addresses:', error);
+      throw error;
+    }
+  }
+
+  /**
+   * Check if an address belongs to this wallet
+   */
+  async isAddressMine(address: string): Promise<boolean> {
+    if (!this.wallet?.isReady()) {
+      throw new Error('Wallet not initialized');
+    }
+
+    try {
+      return await this.wallet.isAddressMine(address);
+    } catch (error) {
+      log.error('Failed to check if address is mine:', error);
+      // Re-throw instead of returning false - calling code needs to distinguish
+      // between "not mine" (false) and "failed to check" (error)
+      throw error;
+    }
+  }
+
+  /**
+   * Get tokens in the wallet
+   */
+  async getTokens(): Promise<Array<Record<string, unknown>>> {
+    if (!this.wallet?.isReady()) {
+      throw new Error('Wallet not initialized');
+    }
+
+    try {
+      const tokens = await this.wallet.getTokens();
+      // wallet-lib returns string[] but we need Record<string, unknown>[]
+      return tokens as unknown as Array<Record<string, unknown>>;
+    } catch (error) {
+      log.error('Failed to get tokens:', error);
+      throw error;
+    }
+  }
+
+  /**
+   * Stop the wallet and cleanup resources
+   */
+  async stop(): Promise<void> {
+    if (!this.wallet) return;
+
+    try {
+      this.wallet.removeAllListeners();
```

Comment:
The `stop` method of the wallet already calls this

Reply:
Okay, removed